cJSON.h

```diff
@@ -81,85 +81,122 @@ typedef struct cJSON_Hooks
       void (*free_fn)(void *ptr);
 } cJSON_Hooks;
 
+#if !defined(__WINDOWS__) && (defined(WIN32) || defined(WIN64) || defined(_MSC_VER) || defined(_WIN32))
+#define __WINDOWS__
+#endif
+#ifdef __WINDOWS__
+
+/* When compiling for windows, we specify a specific calling convention to avoid issues where we are being called from a project with a different default calling convention.  For windows you have 2 define options:
+
+CJSON_DECLARE_STATIC - Define this in the case where you don't want to ever dllexport symbols
```

Comment:
it is not directly related to building static or dynamic, it has to do with which symbols will be exported from the dll.  if you set that define, it would export no symbols, but lets say you built static then linked it into another dll, you could use the symbols inside the dll, but would not be able to use them on the outside of the dll.  agree with the name change.